### DISCUSSION TITLE
question

### DISCUSSION THREAD

**Discussion by FURUNZ** [SeisSol/SeisSol#1010] [Q&A] [answered]
Using the four solvers SeisSol_proxy_Release_dhsw_4_elastic, SeisSol_proxy_Release_dskx_4_elastic, SeisSol_Release_dhsw_4_elastic, and SeisSol_Release_dskx_4_elastic, what are the differences between them? Where can I find detailed information about these solvers, and which one should I use for running different examples?"

**Comment by davschneller** [comment]
Hi, thank you for your question.

I'll convert (or rather: have converted) this issue to a discussion, and answer it there (it's more suited for that style of questions).

**Comment by davschneller** [comment, marked as answer]
Answer: `SeisSol_Release_dhsw_4_elastic` (maybe `SeisSol_Release_dskx_4_elastic` works as well)

More details: the four executables differ in two ways:

* the two of them start with `SeisSol_proxy` (the first two names you mentioned) are used for benchmarking the application, but not for running the examples. You can see that e.g. when running them in the terminal without arguments. In contrast, the latter two of them can be used to run the examples, e.g. by running `SeisSol_Release_dhsw_4_elastic parameters.par` after generating a mesh (or using the precomputed-seissol repository)
* two of the executables were compiled for the target `hsw`, and two for `skx`. The difference here is: `skx` is usually the more efficient/capable one, but it usually only works on compute clusters (it needs AVX512 support). In contrast, `hsw` runs on almost all personal computers after about 2014. I'd suggest to try running `hsw` (i.e. `SeisSol_Release_dhsw_4_elastic`). If everything works, you can try the `skx` (i.e.  `SeisSol_Release_dskx_4_elastic`)—but you may receive a `SIGILL` error when running on a machine which is not capable of it.

Other than that, the name indicates, that you've compiled `SeisSol`, with `Release` build settings, for double precision (hence the `d`) for either `hsw` or `skx`, with order `4`, for the isotropic `elastic` wave equation.

EDIT: an overview over the models supported by Seissol can be e.g. found in https://seissol.readthedocs.io/en/latest/physical-models.html
